app/controllers/api_controller.rb

```diff
@@ -0,0 +1,133 @@
+class ApiController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :check_token
+
+  def duo_limitations
+    study_ids = parse_duo_limitations_payload(request.raw_post)
+    if study_ids.nil?
+      return render json: {'error': 'Invalid payload'}, status: :unprocessable_entity
+    end
+
+    users = User.where(study_id: study_ids)
+
+    duo_limitations_json = users.map do |user|
+      [user.study_id, duo_limitations_for_user(user)]
+    end.to_h
+
+    render json: duo_limitations_json
+  end
+
+  def not_found
+    render json: {'error': 'Not found'}, status: :not_found
+  end
+
+  private
+
+  #
+  # Example:
+  #
+  #   x = {
+  #     "code": "DUO:1",
+  #     "modifiers": [
+  #       { "code": "DUO:2", "regions": ["US"] },
+  #       { "code": "DUO:3", "regions": ["EU"] }
+  #     ]
+  #   }
+  #
+  #   y = {
+  #     "code": "DUO:1",
+  #     "modifiers": [
+  #       { "code": "DUO:2", "regions": ["AU"] },
+  #       { "code": "DUO:4", "regions": ["NZ"] }
+  #     ]
+  #   }
+  #
+  #   merge(x, y) == {
+  #     "code": "DUO:1",
+  #     "modifiers": [
+  #       { "code": "DUO:2", "regions": ["US", "AU"] }
+  #       { "code": "DUO:3", "regions": ["EU"] }
+  #       { "code": "DUO:4", "regions": ["NZ"] }
+  #     ]
+  #   }
+  #
```

Comment:
This is a neat example because the modifiers in `x` and `y` don't contradict each other when merged. Unfortunately, it's possible to define two mutually contradictory `ConditionalDuoLimitations` whose `condition`s are both satisfied at once. I couldn't devise a good check for this kind of inconsistency. Here's an example:

```json
{
  "duoLimitation": {
    "code": "DUO:0000042",
    "modifiers": [
      { "code": "DUO:0000025", "start": "2023-01-01", "end": "2023-01-12" }
    ]
  },
  "condition": { "consent_question_id": 1, "answer": "yes" }
}
```

```json
{
  "duoLimitation": {
    "code": "DUO:0000042",
    "modifiers": [
      { "code": "DUO:0000025", "start": "2019-11-11", "end": "2019-11-12" }
    ]
  },
  "condition": { "consent_question_id": 1, "answer": "yes" }
}
```

With these two `ConditionalDuoExpression`s in place, this nonsense is what the REST API will return for any user answering "yes" to question `1`:

```json
{
  "code": "DUO:0000042",
  "modifiers": [
    { "code": "DUO:0000025", "start": "2023-01-01", "end": "2023-01-12" }
    { "code": "DUO:0000025", "start": "2019-11-11", "end": "2019-11-12" }
  ]
}
```

It's a case of garbage in, garbage out. If CTRL's administrator defines the DUO codes in a dodgy way, I guess that's on them. But it'd still be nice to have a static check to make sure people don't shoot themselves in the foot.

We could restrict the list of `ConditionalDuoExpression`s such that the overarching DUO code (e.g. `DUO:0000042` in this case) can only be defined once. That'd be a start. But we'd also need to place restrictions on the list of modifiers too. Otherwise someone could just have the dynamically-generated, merged nonsense as their "duoLimitation" in a `ConditionalDuoExpression`.